log.go

```
@@ -1,19 +1,61 @@
 package courier
 
-import "context"
+import (
+	"context"
+	"fmt"
+)
 
 // WithLogger sets the Logger to use for the client.
 func WithLogger(l Logger) ClientOption { return optionFunc(func(o *clientOptions) { o.logger = l }) }
 
 // Logger is the interface that wraps the Info and Error methods.
 type Logger interface {
-	Info(ctx context.Context, msg string, attrs map[string]any)
```

This is a breaking change!

Moreover, fatty interfaces are a pain to work with.
Why not use levelled loggers instead?

i.e. Keep the original logger interface, but ask the users for a `PahoVerbosityLevel` via an `Option`, and use that to initialise the relevant loggers, you can initialise all loggers by default except debug one I think.

---

The users are already passing a logger for courier-go logging and some users are also adding custom info like client id, etc. The idea here is to let that info to propagate instead of initialising a new logger just for paho logs and ask for paho verbosity level. IMO this could introduce confusion as users could expect paho logs to also work when passing the logger in the courier-go client init. What do you think? 

I do not have any specific preferences but trying to brainstorm.

---

> This is a breaking change!
Moreover, fatty interfaces are a pain to work with.

Agreed.

---

Given we are introducing paho logging with the courier-go logger in the new release and new log levels are the need of the hour because of paho logging. IMO it is fair to add these in the Logger interface. What do you say?

---

Given this is a breaking change and ideally a feature which enables paho logging using courier-go logger, I have updated the PR title and would be bumping the minor version.

---

> The users are already passing a logger for courier-go logging and some users are also adding custom info like client id, etc.

I agree with you on this point to use the logger to add client context to logs.